src/postgres/inventory/collection/PgCollectionKey.ts

```diff
@@ -134,10 +135,16 @@ class PgCollectionKey implements CollectionKey<PgClassType.Value, PgCollectionKe
    * @private
    */
   @memoizeMethod
-  private _getSelectLoader (client: Client): DataLoader<PgClassType.Value, PgClassType.Value | null> {
+  private _getSelectLoader (client: Client, resolveInfo: mixed, collectionGqlType: mixed): DataLoader<PgClassType.Value, PgClassType.Value | null> {
     return new DataLoader<PgClassType.Value, PgClassType.Value | null>(
-      async (keys: Array<PgClassType.Value>): Promise<Array<PgClassType.Value | null>> => {
+      async (keysAndStuff: mixed): Promise<Array<PgClassType.Value | null>> => {
```

Comment:
This `DataLoader` thing is definitely an interesting and complex problem…

It would be great if we could scan the GraphQL query in its entirety before executing and get a `ReadOptimizations`, or object of a similar concept, that will then be globally static across the request.